react/features/video-quality/subscriber.js

```diff
@@ -193,43 +195,108 @@ function _updateReceiverVideoConstraints({ getState }) {
     const maxFrameHeight = Math.min(maxReceiverVideoQuality, preferredVideoQuality);
     const { remoteScreenShares } = state['features/video-layout'];
     const { visibleRemoteParticipants } = state['features/filmstrip'];
+    const { flags: { sourceNameSignaling } } = state['features/base/config'];
+    const tracks = state['features/base/tracks'];
 
     const receiverConstraints = {
         constraints: {},
         defaultConstraints: { 'maxHeight': VIDEO_QUALITY_LEVELS.NONE },
         lastN,
-        onStageEndpoints: [],
-        selectedEndpoints: []
+        ...sourceNameSignaling ? { prioritizedSources: [] } : { onStageEndpoints: [] },
+        ...sourceNameSignaling ? { selectedSources: [] } : { selectedEndpoints: [] }
     };
 
-    // Tile view.
-    if (shouldDisplayTileView(state)) {
-        if (!visibleRemoteParticipants?.size) {
-            return;
+    if (sourceNameSignaling) {
+        let visibleRemoteTrackSourceNames;
+
+        if (visibleRemoteParticipants?.size) {
+            visibleRemoteTrackSourceNames = [ ...visibleRemoteParticipants ].reduce((acc, participantId) => {
+                const track = getTrackByMediaTypeAndParticipant(tracks, MEDIA_TYPE.VIDEO, participantId);
+                const sourceName = track?.jitsiTrack?.getSourceName?.();
+
+                if (sourceName) {
+                    acc.push(sourceName);
+                }
+
+                return acc;
+            }, []);
         }
 
-        visibleRemoteParticipants.forEach(participantId => {
-            receiverConstraints.constraints[participantId] = { 'maxHeight': maxFrameHeight };
-        });
+        const largeVideoSourceName = getTrackByMediaTypeAndParticipant(
+            tracks, MEDIA_TYPE.VIDEO,
+            largeVideoParticipantId
+        )?.jitsiTrack?.getSourceName?.();
```

Comment:
there's no need for `?` after the `getSourceName`, if the `jitsiTrack` is defined it should always have the method (and it the code should crash if it doesn't)

Reply:
I'll remove the extra `?`.  I am also going to create a `getTrackSourceNameByMediaTypeAndParticipant` selector to wrap this logic